projects/avodata/AvodataSolverTool.java

```diff
@@ -0,0 +1,143 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Scanner;
+
+import org.apache.commons.math3.analysis.function.Abs;
+import org.apache.commons.math3.linear.Array2DRowRealMatrix;
+import org.apache.commons.math3.linear.DecompositionSolver;
+import org.apache.commons.math3.linear.MatrixUtils;
+import org.apache.commons.math3.linear.RealMatrix;
+import org.apache.commons.math3.linear.RealVector;
+import org.apache.commons.math3.linear.SingularValueDecomposition;
+import org.apache.commons.math3.util.CombinatoricsUtils;
+
+/* This class reads matrix from csv file, ignores the parmeters and calculate
+ * using least squares
+ * To build:
+ * bazel build //projects/avodata:ls-alg_tool
+ * To run:
+ * bazel-bin/projects/avodata/ls-alg_tool
+ */
+
+public class AvodataSolverTool {
+  // Path to csv file
+  static final String CSV_PATH = "/projects/avodata/check.csv";
+
+  public static double factorial(int n) {
+    return CombinatoricsUtils.factorialDouble(n);
+  }
+
+  public static double nChooseK(int n, int k) {
+    return CombinatoricsUtils.binomialCoefficientDouble(n, k);
+  }
+
+  public static RealMatrix generateSyntheticA(int numOfProfessions) {
+    int numOfraws = (int) nChooseK(numOfProfessions, 2);
+    RealMatrix matrix = new Array2DRowRealMatrix(numOfraws, numOfProfessions);
+
+    int row = 0;
+    for (int i = 0; i < numOfProfessions; i++)
+      for (int j = i + 1; j < numOfProfessions; j++) {
+        matrix.addToEntry(row, i, 1);
+        matrix.addToEntry(row, j, -1);
+        row++;
+      }
+    return matrix;
+  }
+
+  public static RealMatrix generateSyntethicC(RealMatrix m, int i, RealMatrix a) {
+    RealMatrix c = a.multiply(m.getRowMatrix(i).transpose());
```

Comment:
You don't need an intermediate variable. You can just return directly:
`return a.multiply(m.getRowMatrix(i).transpose());`